mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,381 @@
+from __future__ import print_function, absolute_import, division
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import statsmodels.api as sm
+import statsmodels.formula.api as smf
+from scipy.stats import chi2
+
+import mousestyles.data as data
+
+
+def aggregate_interval(strain, mouse, feature, bin_width):
+    """
+    Aggregate the interval data based on n-minute time
+    intervals, return a time series.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "M_AS", "M_IS", "W"}
+        "AS": Active state probalibity
+        "F": Food consumed (g)
+        "M_AS": Movement outside homebase
+        "M_IS": Movement inside homebase
+        "W": Water consumed (g)
+    bin_width: number of minutes of time interval for data aggregation
+
+    Returns
+    -------
+    ts: pandas.tseries
+        a pandas time series of length 12(day)*24(hour)*60(minute)/n
+    """
+    # load data
```

Comment:
Add input check for features (in given set). Add input check for strain mouse (positive integer). Add input check for bin_width(min and max range), etc.